internal/error.go

```diff
@@ -1029,3 +1045,42 @@ func getErrType(err error) string {
 
 	return t.Name()
 }
+
+func applicationErrorCategoryToProto(category ApplicationErrorCategory) enumspb.ApplicationErrorCategory {
+	switch category {
+	case ErrorCategoryBenign:
+		return enumspb.APPLICATION_ERROR_CATEGORY_BENIGN
+	case "":
+		// Zero value maps to unspecified
+		return enumspb.APPLICATION_ERROR_CATEGORY_UNSPECIFIED
+	default:
+		// Fallback to unspecified if unknown case
+		return enumspb.APPLICATION_ERROR_CATEGORY_UNSPECIFIED
+	}
+}
+
+func applicationErrorCategoryFromProto(category enumspb.ApplicationErrorCategory) ApplicationErrorCategory {
+	switch category {
+	case enumspb.APPLICATION_ERROR_CATEGORY_BENIGN:
+		return ErrorCategoryBenign
+	case enumspb.APPLICATION_ERROR_CATEGORY_UNSPECIFIED:
+		// Unspecified maps to zero value
+		return ""
+	default:
+		// Fallback to zero value if unknown case
+		return ""
+	}
+}
+
+func IsBenignApplicationError(err error) bool {
```

Comment:
```suggestion
func isBenignApplicationError(err error) bool {
```

No need to expose I assume unless it's a helper we really need to expose, but I don't believe we do. A user can easily check for any value on application error without a helper.

Reply:
Didn't mean to export this, unexported.

Reply:
Did you want this removed btw? Just do the check inline wherever it's used?
Alternatively, should we put this check under a different name (i.e. `shouldRecordFailureMetric` or something similar, probably move it to metrics or something), in case we adopt more conditional logic for failure metrics in the future?